File: src/ui/src/api/stats.ts
import { api } from './client';

export interface DailyStat {
  day: string; // "YYYY-MM-DD"
  ms_listened: number;
}

export interface FinishedBook {
  book_id: number;
  times_finished: number;
  finished_at: string;
}

export async function getDailyStats(days = 30): Promise<DailyStat[]> {
  return api.get<DailyStat[]>(`/stats/daily?days=${days}`);
}

export async function getFinishedBooks(): Promise<FinishedBook[]> {
  return api.get<FinishedBook[]>('/stats/finished');
}

File: src/models/stats.rs
use chrono::{DateTime, Utc};
use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct DailyStat {
    pub day: String,
    pub ms_listened: i64,
}

#[derive(Debug, sqlx::FromRow, Serialize)]
pub struct FinishedBook {
    pub book_id: i64,
    pub times_finished: i64,
    pub finished_at: DateTime<Utc>,
}
